src/activities/solar_system/solar_system.js

```diff
@@ -0,0 +1,112 @@
+/* GCompris - solar_system.js
+ *
+ * Copyright (C) 2017 Aman Kumar Gupta <gupta2140@gmail.com>
+ * Authors:
+ *   Aman Kumar Gupta <gupta2140@gmail.com>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import QtQuick 2.6 as Quick
+.import "Dataset.js" as Dataset
+.import "qrc:/gcompris/src/core/core.js" as Core
+
+var currentLevel = 0
+var numberOfLevel
+var items
+var dataset
+var currentPlanetLevels
+var currentSubLevel
+var indexOfSelectedPlanet
+
+function start(items_) {
+    items = items_
+    currentLevel = 0
+    dataset= Dataset.get()
+    for(var i = 0;  i < dataset.length; ++i) {
+        items.containerModel.append({
+                "clipImg": dataset[i].clipImg,
+                "bodyName": dataset[i].bodyName
+        });
+    }
+    items.solarSystemVisible = true
+}
+
+function stop() {
+}
+
+function initLevel() {
+    currentSubLevel = 0
+    items.bar.level = currentLevel + 1
+    items.mainQuizScreen.score.numberOfSubLevels = currentPlanetLevels.length
+    Core.shuffle(currentPlanetLevels)
+    nextSubLevel();
+}
+
+function showSolarModel() {
+    currentLevel = 0
+    items.quizScreenVisible = false
+    items.solarSystemVisible = true
+}
+
+function showQuizScreen(index) {
+    indexOfSelectedPlanet = index
+    items.solarSystemVisible = false
+    items.quizScreenVisible = true
+    currentPlanetLevels = dataset[index].levels
+    items.mainQuizScreen.planetRealImage = dataset[index].realImg
+    items.hintDialog.hint1 = "1. The <b>farther</b> a planet from the Sun, the <b>lower</b> is it's temperature.<br><font color=\"#3bb0de\">%1</font>".arg(dataset[indexOfSelectedPlanet].temperatureHint)
```

Comment:
qsTr()
Is this text displayed? (we always use the one from the dataset?)

Reply:
Yes, this text is displayed along with the ones in the dataset.
The text here, tells us the proportionate variation of the properties among planets (common for all the planets), and the ones in the dataset gives the datas which can be used as reference values to reach the answer(differ with planets).

Reply:
Then, it misses the qsTr() :)

Reply:
yes, I had added it along with all the other changes in the reviews, yesterday in the commits below ;)